### PR TITLE
MINOR: Small cleanups in connect:runtime

### PR DESCRIPTION
**Pull request by mimaison**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaStatusBackingStore.java [hunk 1]
```diff
@@ -137,11 +137,6 @@
     private int generation;
     private SharedTopicAdmin ownTopicAdmin;
 
-    @Deprecated
```
Comment (hachikuji) [inline connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaStatusBackingStore.java]: Do you know why we had this deprecated? Do we consider it a public API?

Below we have code which checks for null `SharedTopicAdmin`. Seems like we can probably get rid of that. The only usage I could see was in a test case.

**Review by hachikuji — COMMENTED** [review]
Looks good overall. Left a minor comment.

File: connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaStatusBackingStore.java (hunk 1 above)
Comment (mimaison) [reply connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaStatusBackingStore.java]: It's not public API but looking at the history it looks like we're aware of external users and we decided to keep compatibility. For that reason, I've reverted this change.

**Review by hachikuji — APPROVED** [review]
Thanks, LGTM

**Pull request merged**